client/app/hosting/framework/envvar/add/hosting-framework-envvar-add.html

```diff
@@ -0,0 +1,59 @@
+
+        <div data-wizard-step
+             data-wizard-step-valid="isValid()">
+
+            <form class="form-horizontal" name="addEnvvarForm" data-ng-if="!ctrl.loading">
+                <p>
+                    <small class="text-danger">*</small>
+                    <small data-i18n-static="required_fields"></small>
+                </p>
+
+                <div class="form-group"
+                     data-ng-class="{ 'has-error': ctrl.entryToCreate.key && !ctrl.isKeyValid() }">
+                    <label class="control-label col-md-4 required" for="envvarKey" data-i18n-static="hosting_tab_FRAMEWORK_envvar_label_key"></label>
+
+                    <div class="col-md-6">
+                        <input type="text" class="form-control" id="envvarKey" data-ng-model="ctrl.entryToCreate.key">
+                        <small class="help-block" role="alert"
+                               data-i18n-static="hosting_tab_FRAMEWORK_envvar_label_key_error"
+                               data-ng-if="ctrl.entryToCreate.key && !ctrl.isKeyValid()">
+                        </small>
+                    </div>
+                </div>
+
+                <div class="form-group">
+                    <label class="control-label col-md-4 required" for="envvarType" data-i18n-static="hosting_tab_FRAMEWORK_envvar_label_type"></label>
+
+                    <div class="col-md-8">
+                        <div class="oui-select oui-select_inline mb-0">
+                            <select class="oui-select__input" id="envvarType" required
+                                    data-ng-change="ctrl.entryToCreate.value = null"
+                                    data-ng-model="ctrl.entryToCreate.type">
+                                <option>string</option>
+                                <option>integer</option>
+                                <option>password</option>
+                            </select>
+
+                            <span class="oui-icon oui-icon-chevron-down" aria-hidden="true"></span>
+                        </div>
+                    </div>
+                </div>
+
+                <div class="form-group">
+                    <label class="control-label col-md-4 required" for="envvarValue" data-i18n-static="hosting_tab_FRAMEWORK_envvar_label_value"></label>
+
+                    <div class="col-md-6">
+                        <input type="number" data-ng-if="ctrl.entryToCreate.type === 'integer'" class="form-control" id="envvarValue" data-ng-model="ctrl.entryToCreate.value">
+                        <input type="password" data-ng-if="ctrl.entryToCreate.type === 'password'" class="form-control" id="envvarValue" data-ng-model="ctrl.entryToCreate.value">
+                        <input type="text" data-ng-if="ctrl.entryToCreate.type !== 'password' && ctrl.entryToCreate.type !== 'integer'" class="form-control" id="envvarValue" data-ng-model="ctrl.entryToCreate.value">
```

Comment:
Isn't there only 3 types ? `password`, `integer` and `text' ? 

Also, it's required, so it should have the `required` attribute.

Are all values allowed ?